examples/read_nodes_serde.rs

```diff
@@ -0,0 +1,102 @@
+// note: to use serde, the feature needs to be enabled
+// run example with:
+//    cargo run --example read_nodes_serde --features="serialize"
+
+use quick_xml::de::DeError;
+use quick_xml::de::Deserializer;
+use serde::Deserialize;
+
+#[derive(Debug, PartialEq, Default, Deserialize)]
+#[serde(default)]
+struct Translation {
+    #[serde(rename = "@Tag")]
+    tag: String,
+    #[serde(rename = "@Language")]
+    lang: String,
+    #[serde(rename = "Text")]
+    text: String,
+}
+
+#[derive(Debug, PartialEq, Default, Deserialize)]
+#[serde(default)]
+struct DefaultSettings {
+    #[serde(rename = "@Language")]
+    language: String,
+    #[serde(rename = "@Greeting")]
+    greeting: String,
+}
+
+#[derive(Debug, PartialEq, Default, Deserialize)]
+#[serde(default)]
+struct Config {
+    #[serde(rename = "DefaultSettings")]
+    settings: DefaultSettings,
+    #[serde(rename = "Localization")]
+    translation: Vec<Translation>,
+}
+
+const XML: &str = r#"
+<?xml version="1.0" encoding="utf-8"?>
+<Config>
+  <DefaultSettings Language="es" Greeting="HELLO"/>
+  <Localization>
+    <Translation Tag="HELLO" Language="ja">
+      <Text>こんにちは</Text>
+    </Translation>
+    <Translation Tag="BYE" Language="ja">
+      <Text>さようなら</Text>
+    </Translation>
+    <Translation Tag="HELLO" Language="es">
+      <Text>Hola</Text>
+    </Translation>
+    <Translation Tag="BYE" Language="es">
+      <Text>Adiós</Text>
+    </Translation>
+  </Localization>
+</Config>
+"#;
+
+const ONE_TRANSLATION_XML: &str = r#"
+    <Translation Tag="HELLO" Language="ja">
+      <Text>こんにちは</Text>
+    </Translation>
+"#;
+
+/// Deserialize an instance of type T from a string of XML text.
+/// If deserialization was succeeded checks that all XML events were consumed
+fn from_str<'de, T>(s: &'de str) -> Result<T, DeError>
+where
+    T: Deserialize<'de>,
+{
+    let mut de = Deserializer::from_str(s);
+    let result = T::deserialize(&mut de);
+
+    // If type was deserialized, the whole XML document should be consumed
+    if let Ok(_) = result {
+        match <()>::deserialize(&mut de) {
+            Err(DeError::UnexpectedEof) => (),
+            e => panic!("Expected end `UnexpectedEof`, but got {:?}", e),
+        }
+    }
+
+    result
+}
```

Comment:
That is a special reimplementation of [`from_str`](https://docs.rs/quick-xml/latest/quick_xml/de/fn.from_str.html) for tests, to ensure that we consume all input, no need to use it in an example. Just use [`from_str`](https://docs.rs/quick-xml/latest/quick_xml/de/fn.from_str.html).

Reply:
ah, thank you!